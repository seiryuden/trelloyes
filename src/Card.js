import React from 'react';
import "./Card.css"

function Card(props){
    
    
    return (
        <div className="Card">
            <h1>{props.cardId}</h1>
            <h3>{props.title}</h3>
            <p>{props.content}</p>
            <button onClick={() => props.onDeleteItem(props.listKey, props.cardId)}>Delete</button>
            
        </div>
    );
}

export default Card;