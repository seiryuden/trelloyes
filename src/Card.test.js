import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Card from "./Card";

describe("Card component", ()=> {

it ("renders without crashing", () =>{

    const div = document.createElement("div");
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);

});

it ("renders the UI as expected", () => {

    const tree = renderer
        .create(<Card key={0} title="Third Card" content="this is the content"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

});