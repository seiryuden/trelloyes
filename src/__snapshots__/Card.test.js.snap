// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <h3>
    Third Card
  </h3>
  <p>
    this is the content
  </p>
</div>
`;
